ush/parsing_namelists_FV3.sh

```diff
@@ -0,0 +1,603 @@
+  io_layout = $io_layout
+  npx = $npx
+  npy = $npy
+  ntiles = $ntiles
+  npz = $npz
+EOF
+
+if [ $cpl = .true. ]; then
+  cat >> input.nml << EOF
+  dz_min =  ${dz_min:-"6"}    ! no longer in develop branch
+  psm_bc = ${psm_bc:-"0"}    ! no longer in develop branch
+EOF
+fi
+
+cat >> input.nml << EOF
+  grid_type = -1
+  make_nh = $make_nh 
+  fv_debug = ${fv_debug:-".false."}
+  range_warn = ${range_warn:-".false."}
+  reset_eta = .false.
+  n_sponge = ${n_sponge:-"10"}    
+  nudge_qv = ${nudge_qv:-".true."}
+  nudge_dz = ${nudge_dz:-".false."}
+  tau = ${tau:-10.}
+  rf_cutoff = ${rf_cutoff:-"7.5e2"}
+  d2_bg_k1 = ${d2_bg_k1:-"0.15"}    
+  d2_bg_k2 = ${d2_bg_k2:-"0.02"}    
+  kord_tm = ${kord_tm:-"-9"}
+  kord_mt = ${kord_mt:-"9"}
+  kord_wz = ${kord_wz:-"9"}
+  kord_tr = ${kord_tr:-"9"}
+  hydrostatic = $hydrostatic
+  phys_hydrostatic = $phys_hydrostatic
+  use_hydro_pressure = $use_hydro_pressure
+  beta = 0.
+  a_imp = 1.
+  p_fac = 0.1
+  k_split = $k_split
+  n_split = $n_split
+  nwat = ${nwat:-2}	
+  na_init = $na_init
+  d_ext = 0.
+  dnats = ${dnats:-0}		
+  fv_sg_adj = ${fv_sg_adj:-"450"}
+  d2_bg = 0.
+  nord = ${nord:-3}		
+  dddmp = ${dddmp:-0.2}	
+  d4_bg = ${d4_bg:-0.15}
+  vtdm4 = $vtdm4
+  delt_max = ${delt_max:-"0.002"}
+  ke_bg = 0.
+  do_vort_damp = $do_vort_damp
+  external_ic = $external_ic
+  external_eta = ${external_eta:-.true.}
+  gfs_phil = ${gfs_phil:-".false."}
+  nggps_ic = $nggps_ic
+  mountain = $mountain
+  ncep_ic = $ncep_ic
+  d_con = $d_con
+  hord_mt = $hord_mt	
+  hord_vt = $hord_xx
+  hord_tm = $hord_xx
+  hord_dp = -$hord_xx		
+  hord_tr = ${hord_tr:-"8"}	
+  adjust_dry_mass = ${adjust_dry_mass:-".true."}
+  dry_mass=${dry_mass:-98320.0}
+  consv_te = $consv_te
+  do_sat_adj = ${do_sat_adj:-".false."}	
+  consv_am = .false.
+  fill = .true.
+  dwind_2d = .false.
+  print_freq = $print_freq
+  warm_start = $warm_start
+  no_dycore = $no_dycore
+  z_tracer = .true.
+  agrid_vel_rst = ${agrid_vel_rst:-".true."}
+  read_increment = $read_increment
+  res_latlon_dynamics = $res_latlon_dynamics
+  $fv_core_nml
+/
+
+&external_ic_nml
+  filtered_terrain = $filtered_terrain
+  levp = $LEVS
+  gfs_dwinds = $gfs_dwinds
+  checker_tr = .false.
+  nt_checker = 0
+  $external_ic_nml
+/
+
+&gfs_physics_nml
+  fhzero       = $FHZER
+  h2o_phys     = ${h2o_phys:-".true."}	
+  ldiag3d      = ${ldiag3d:-".false."}
+  fhcyc        = $FHCYC			
+  use_ufo      = ${use_ufo:-".true."}
+  pre_rad      = ${pre_rad:-".false."}
+  imp_physics  = ${imp_physics:-"99"}	
+EOF
+
+case "${CCPP_SUITE:-}" in
+  "FV3_GFS_v15p2_coupled")
+  cat >> input.nml << EOF
+  oz_phys      = .false.
+  oz_phys_2015 = .true.
+EOF
+  ;;
+  "FV3_GSD_v0")
+  cat >> input.nml << EOF
+  iovr         = ${iovr:-"3"}
+  ltaerosol    = ${ltaerosol:-".F."}    
+  lradar       = ${lradar:-".F."}	
+  ttendlim     = ${ttendlim:-0.005}
+  oz_phys      = ${oz_phys:-".false."}
+  oz_phys_2015 = ${oz_phys_2015:-".true."}
+  lsoil_lsm    = ${lsoil_lsm:-"4"}
```

Comment:
Currently only FV3_GFS_v16* and FV3_GSD_v0 suite have lsoil_lsm in place.

Reply:
This might need to be moved elsewhere. I remember removing it from coupled because it was no longer recognized, maybe because of Noah-MP? I'd have to go back and look.